docs/pages/omnia3/Modeler/omnia3_modeler_components.md

```diff
@@ -0,0 +1,189 @@
+---
+title: Components
+keywords: lowcode user interface components
+summary: "All the information regarding our User Interface Components. Use your own Components on OMNIA Pages"
+sidebar: omnia3_sidebar
+permalink: omnia3_modeler_components.html
+folder: omnia3
+---
+
+## 1. Introduction
+
+On OMNIA Platform models, you can add your own Components to be used on [Pages](omnia3_modeler_pages.md).
+
+These components have the same purpose as the ones available on Forms and Dashboards but, unlike those, they can be fully controlled by the modeler.
+
+Components can be visual (e.g. inputs) or functional. Users can interact with the visual components directly, while functional components are used to execute multiple processes behind the screen (e.g. fetch data, act as container to render other elements).
+
+OMNIA Platform contains a small number of internal components, and new ones can be added to each model, according to its needs. Multiple packages with different components can be uploaded, and you can develop your own components. To start, we suggest using the package made available by the OMNIA Platform, that you can find [here](https://github.com/OMNIALowCode/omnia-base-components).
+
+## 2. Components Structure
+
+On the modeler perspective, a Component is composed by the following definitions:
+
+### Properties
+
+The component attributes contain data relevant to define the component and change how it renders and behaves on a page. A component can have multiple attributes, and they can be different between components.
+
+An attribute can be required, and have a default value to be suggested when modeling. Depending on the attribute _direction_ configuration, its value can be only read (Outbound direction) or changed (Inbound direction), or both (TwoWay direction).
```

Comment:
The attribute can be defined as Outbound, Inbound, and TwoWay. Outbound attributes are read-only and only the component can change the value. Inbound attributes can only be changed by Page "interaction". TwoWay attributes can be changed for all.